File: lab-03/by/LEXUS_FAMCS/paint/Loader.java
package by.LEXUS_FAMCS.paint;

import javafx.scene.canvas.Canvas;
import javafx.scene.canvas.GraphicsContext;
import javafx.scene.image.Image;
import javafx.stage.FileChooser;

import java.io.File;
import java.io.IOException;

public class Loader {
    DataModel dataModel;
    private Canvas canvas;

    Loader(Canvas canvas, DataModel dataModel) {
        this.canvas = canvas;
        this.dataModel = dataModel;
    }

    public File load() {
        FileChooser fileChooser = new FileChooser();
        fileChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("PNG files (*.png)", "*.png"));
        fileChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("JPG files (*.jpg)", "*.jpg"));
        fileChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("JPEG files (*.jpeg)", "*.jpeg"));
        fileChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("GIF files (*.gif)", "*.gif"));
        fileChooser.getExtensionFilters().add(
                new FileChooser.ExtensionFilter("Image files (*.png, *.jpeg, *.jpg, *.gif)", "*.png", "*.jpeg", "*.jpg", "*.gif")
        );
        File file = fileChooser.showOpenDialog(null);

        if (file != null) {
            try {
                dataModel.toFile = file;
                loadFromFile(file);
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
        return file;
    }

    private void loadFromFile(File file) throws IOException {
        Image image = new Image(file.toURI().toString());
        GraphicsContext gc = canvas.getGraphicsContext2D();
        gc.clearRect(0, 0, canvas.getWidth(), canvas.getHeight());
        gc.drawImage(image, 0, 0, canvas.getWidth(), canvas.getHeight());
    }
}

File: lab-03/by/Kra567/paint/Loader.kt
package by.Kra567.paint

import javafx.embed.swing.SwingFXUtils
import javafx.event.EventHandler
import javafx.scene.canvas.Canvas
import javafx.scene.control.Button
import javafx.scene.image.Image
import javafx.stage.FileChooser
import javafx.stage.Stage
import java.awt.event.ActionEvent
import java.io.File
import java.io.FileInputStream
import javax.imageio.ImageIO
import kotlin.math.max

class Loader(private var stage : Stage,private var canvas : ActedCanvas) : Button("load") {
    init{
        maxWidth = Double.POSITIVE_INFINITY
        onAction = EventHandler { load() }
    }
    private fun load(){
        val fileChooser = FileChooser()
        fileChooser.extensionFilters.add(FileChooser.ExtensionFilter("image formats","*.jpg","*.png"))
        val file : File? = fileChooser.showOpenDialog(null)
        file?.let{
            stage.isResizable = true
            val fileInputStream = FileInputStream(file)
            val image: Image = SwingFXUtils.toFXImage(ImageIO.read(fileInputStream), null)






            fileInputStream.close()
            stage.width = stage.width - canvas.width + image.width
            stage.height = max(image.height,stage.height)
            canvas.width = image.width
            canvas.height = image.height
            canvas.graphicsContext2D.drawImage(image, 0.0, 0.0)
            stage.isResizable = false
        }

    }
}